File: src/main/kotlin/ninja/blacknet/core/HTLC.kt
/*
 * Copyright (c) 2018-2019 Pavel Vasin
 *
 * Licensed under the Jelurida Public License version 1.1
 * for the Blacknet Public Blockchain Platform (the "License");
 * you may not use this file except in compliance with the License.
 * See the LICENSE.txt file at the top-level directory of this distribution.
 */

package ninja.blacknet.core

import kotlinx.serialization.Serializable
import ninja.blacknet.contract.HashLock
import ninja.blacknet.contract.TimeLock
import ninja.blacknet.crypto.PublicKey

@Serializable
class HTLC(
        val height: Int,
        val time: Long,
        val amount: Long,
        val from: PublicKey,
        val to: PublicKey,
        val timeLock: TimeLock,
        val hashLock: HashLock
) {

}

File: kernel/src/main/rust/htlc.rs
/*
 * Copyright (c) 2018-2025 Pavel Vasin
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program. If not, see <https://www.gnu.org/licenses/>.
 */

use crate::amount::Amount;
use crate::ed25519::PublicKey;
use crate::hashlock::HashLock;
use crate::timelock::TimeLock;
use blacknet_time::Seconds;
use serde::{Deserialize, Serialize};

#[derive(Deserialize, Serialize)]
pub struct HTLC {
    pub height: u32,
    pub time: Seconds,
    pub amount: Amount,
    pub from: PublicKey,
    pub to: PublicKey,
    pub time_lock: TimeLock,
    pub hash_lock: HashLock,
}
